p2p/peer.go

```diff
@@ -293,6 +342,10 @@ type PeerManagerOptions struct {
 	// necessary to make room for these.
 	PersistentPeers []NodeID
 
+	// MaxPeers is the maximum number of peers to track. When this is exceeded,
+	// the lowest-scored unconnected peers will get removed. 0 means no limit.
+	MaxPeers uint16
```

Comment:
How does this config relate to `MaxConnected`? Will this value typically be larger than `MaxConnected`? What happens when it's smaller?

Reply:
This is the maximum number of peers that we track (know about) and store in the peer store. It needs to be bounded, since an adversary can otherwise just keep connecting to us with new peer IDs and OOM the node. It will typically be much larger than `MaxConnected`, e.g. 1000.

If `MaxPeers` is less than `MaxConnected` then we'll only connect to `MaxPeers` peers and have a bunch of free connection slots. We can't connect to peers we don't know about.

Reply:
Gotcha. Can we ensure via validation the invariant that `MaxPeers >= MaxConnected`?

Reply:
Sure thing, will add `PeerManagerOptions` validation in a separate PR. Note that there will also be another layer of validation in `config.P2PConfig` which is independent of this (i.e. we probably don't want to expose all of these options all the way out to end users, some of them can be hardcoded -- but still useful to vary in e.g. tests).